File: shared/src/main/java/com/screentime/shared/model/Family.kt
package com.screentime.shared.model

/** A member's role within a family. */
enum class FamilyRole {
    /** The main admin (family creator) plus anyone they promote. Manages members. */
    ADMIN,

    /** A regular co-parent: can view and operate, but can't manage members or the TV. */
    USER,
}

/**
 * A family and its members. The [ownerUid] is the single "main admin" — the
 * only account allowed to pair/unpair the TV and delete the family.
 */
data class Family(
    val id: String,
    val ownerUid: String,
    val members: Map<String, FamilyRole>,
    val deviceIds: List<String>,
) {
    fun roleOf(uid: String): FamilyRole? = members[uid]
    fun isOwner(uid: String): Boolean = uid == ownerUid
    fun isAdmin(uid: String): Boolean = members[uid] == FamilyRole.ADMIN
}

File: web/src/models/Family.ts
export type FamilyRole = 'ADMIN' | 'USER';

/**
 * A family and its members. `ownerUid` is the single "main admin" — the only
 * account allowed to pair/unpair the TV and delete the family.
 */
export interface Family {
  id: string;
  ownerUid: string;
  members: Record<string, FamilyRole>;
  deviceIds: string[];
}

export function roleOf(family: Family, uid: string): FamilyRole | undefined {
  return family.members[uid];
}

export function isOwner(family: Family, uid: string): boolean {
  return uid === family.ownerUid;
}

export function isAdmin(family: Family, uid: string): boolean {
  return family.members[uid] === 'ADMIN';
}
